addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/steps/OpenShiftSteps.java

```diff
@@ -2,39 +2,74 @@
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
 
 import javax.enterprise.context.RequestScoped;
 import javax.inject.Inject;
 
+import io.fabric8.kubernetes.api.KubernetesHelper;
 import io.fabric8.launcher.osio.Annotations;
 import io.fabric8.launcher.osio.che.CheStack;
 import io.fabric8.launcher.osio.che.CheStackDetector;
 import io.fabric8.launcher.osio.projectiles.OsioProjectile;
-import io.fabric8.launcher.service.git.api.GitService;
+import io.fabric8.launcher.osio.tenant.Namespace;
+import io.fabric8.launcher.osio.tenant.Tenant;
 import io.fabric8.launcher.service.openshift.api.OpenShiftService;
+import io.fabric8.openshift.api.model.Build;
+import io.fabric8.openshift.api.model.BuildRequest;
+import io.fabric8.openshift.api.model.BuildRequestBuilder;
+import io.fabric8.openshift.client.OpenShiftClient;
 
 /**
  * @author <a href="mailto:ggastald@redhat.com">George Gastaldi</a>
  */
 @RequestScoped
 public class OpenShiftSteps {
 
-    @Inject
-    private GitService gitService;
+    private static Logger log = Logger.getLogger(OpenShiftSteps.class.getName());
 
     @Inject
     private OpenShiftService openShiftService;
 
+    @Inject
+    private Tenant tenant;
+
+
     public void createBuildConfig(OsioProjectile projectile) {
         Map<String, String> annotations = getBuildConfigAnnotations(projectile);
 
     }
 
+
+    public void triggerBuild(OsioProjectile projectile) {
```

Comment:
wasn't the plan to create the webhooks first and then push the code so that we don't have to trigger a build manually but let the webhook take care of that?

Reply:
Yes, that method was just a first step in the refactoring process, feel free to remove it as you like